sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamBuiltinAggregations.java

```diff
@@ -383,4 +392,30 @@ public Long extractOutput(Long accum) {
       return accum;
     }
   }
+
+  static class BitAnd<T extends Number> extends CombineFn<T, Long, Long> {
+    @Override
+    public Long createAccumulator() {
+      return -1L;
```

Comment:
-1L makes it a bit harder to read. Why not use 1L instead?

Reply:
Because -1L is represented as 64 bits of 1 in binary while 1L only has one 1 at the least significant bit. 

In order to do bit_and operation, I think the initial bit mask should consists of only 1s, so I use -1L.

Reply:
That's a good point 👍  I should go back to re-visit CMU 15231 course slides.

Reply:
Thanks!

BTW I want to mention that I also added 1 line at the end of the previous testBitOrFunction(). I noticed that without this line, the previous bit_or test will always no matter what expected result I set, which makes it an invalid test. 

It will be good if you can take a look.

Reply:
O for that pipeline run and then wait. Thanks that's a nice catch.